yarn-project/end-to-end/src/integration_l1_publisher.test.ts

```diff
@@ -371,22 +395,44 @@ describe('L1Publisher integration', () => {
       const expectedData = encodeFunctionData({
         abi: RollupAbi,
         functionName: 'process',
-        args: [`0x${l2Proof.toString('hex')}`, `0x${block.toBufferWithLogs().toString('hex')}`],
+        args: [
+          `0x${block.header.toBuffer().toString('hex')}`,
+          `0x${block.archive.toBuffer().toString('hex')}`,
+          `0x${block.bodyToBuffer().toString('hex')}`,
+          `0x${l2Proof.toString('hex')}`,
+        ],
       });
       expect(ethTx.input).toEqual(expectedData);
 
-      const decoderArgs = [`0x${block.toBufferWithLogs().toString('hex')}`] as const;
-      const decodedHashes = await decoderHelper.read.computeDiffRootAndMessagesHash(decoderArgs);
-      const decodedRes = await decoderHelper.read.decode(decoderArgs);
-      const stateInRollup = await rollup.read.rollupStateHash();
+      const headerDecoderArgs = [`0x${block.header.toBuffer().toString('hex')}`] as const;
+      const decodedHeader = await headerDecoderHelper.read.decode(headerDecoderArgs);
+
+      expect(Number(block.header.globalVariables.chainId.toBigInt())).toEqual(
+        Number(decodedHeader.globalVariables.chainId),
+      );
+      expect(Number(block.header.globalVariables.version.toBigInt())).toEqual(
+        Number(decodedHeader.globalVariables.version),
+      );
+      expect(block.number).toEqual(Number(decodedHeader.globalVariables.blockNumber));
+      expect(Number(block.header.globalVariables.timestamp.toBigInt())).toEqual(
+        Number(decodedHeader.globalVariables.timestamp),
+      );
+
+      // TODO(benesjan): this ugly. Unify how the archive value is represented.
```

Comment:
Added a comment earlier, but we can do some struct in sol to also have the snapshots.

Reply:
Addressed in https://github.com/AztecProtocol/aztec-packages/pull/4015/commits/02e5359fad01667a78d2e64259b05b59ee17986e and in https://github.com/AztecProtocol/aztec-packages/pull/4015/commits/8a886836ac04b727c5ec4af1703753c7caf3d7cf